mod/notifications/views/default/forms/notificationsettings/groupsave.php

```diff
@@ -27,85 +27,78 @@
+$body .= elgg_format_element('div', [], elgg_echo('notifications:subscriptions:groups:description'));
 
-<div class="elgg-module elgg-module-info">
-	<div class="elgg-body">
-	<?php
-		echo elgg_view('notifications/subscriptions/jsfuncs',$vars);
-	?>
-		<div>
-		<?php
-			echo elgg_echo('notifications:subscriptions:groups:description');
-		?>
-		</div>
-<?php
-
-if (isset($vars['groups']) && !empty($vars['groups'])) {
+$groups = elgg_extract('groups', $vars);
+if ($groups) {
 
-?>
-		<table id="notificationstable" cellspacing="0" cellpadding="4" width="100%">
-			<tr>
-				<td>&nbsp;</td>
-<?php
+	$top_row = elgg_format_element('td', [], '&nbsp;');
 
-	$i = 0; 
+	$i = 0;
 	foreach($NOTIFICATION_HANDLERS as $method => $foo) {
 		if ($i > 0) {
-			echo "<td class='spacercolumn'>&nbsp;</td>";
+			$top_row .= elgg_format_element('td', ['class' => 'spacercolumn'], '&nbsp;');
 		}
-?>
-				<td class="<?php echo $method; ?>togglefield"><?php echo elgg_echo('notification:method:'.$method); ?></td>
-<?php
+		
+		$top_row .= elgg_format_element('td', ['class' => "{$method}togglefield"], elgg_echo("notification:method:{$method}"));
 		$i++;
 	}
-?>
-				<td>&nbsp;</td>
-			</tr>
-<?php
-	foreach($vars['groups'] as $group) {
+	$top_row .= elgg_format_element('td', [], '&nbsp;');
+	
+	$table_data = elgg_format_element('tr', [], $top_row);
+
+	foreach($groups as $group) {
 		
 		$fields = '';
 		$i = 0;
 		
 		foreach($NOTIFICATION_HANDLERS as $method => $foo) {
-			if (in_array($group->guid,$subsbig[$method])) {
-				$checked[$method] = 'checked="checked"';
-			} else {
-				$checked[$method] = '';
-			}
+			
 			if ($i > 0) {
-				$fields .= "<td class=\"spacercolumn\">&nbsp;</td>";
+				$fields .= elgg_format_element('td', ['class' => 'spacercolumn'], '&nbsp;');
 			}
-			$fields .= <<< END
-				<td class="{$method}togglefield">
-				<a border="0" id="{$method}{$group->guid}" class="{$method}toggleOff" onclick="adjust{$method}_alt('{$method}{$group->guid}');">
-				<input type="checkbox" name="{$method}subscriptions[]" id="{$method}checkbox" onclick="adjust{$method}('{$method}{$group->guid}');" value="{$group->guid}" {$checked[$method]} /></a></td>
-END;
+			
+			$toggle_input = elgg_view('input/checkbox', [
+				'name' => "{$method}subscriptions[]",
+				'id' => "{$method}checkbox",
+				'value' => $group->guid,
+				'checked' => in_array($group->guid, $subsbig[$method]),
+				'onclick' => "adjust{$method}('{$method}{$group->guid}');",
+			]);
+			$toggle_link = elgg_view('output/url', [
+				'href' => false,
+				'text' => $toggle_input,
+				'id' => "{$method}{$group->guid}",
+				'class' => "{$method}toggleOff",
+				'border' => '0',
+				'onclick' => "adjust{$method}_alt('{$method}{$group->guid}');",
+			]);
+			
+			$fields .= elgg_format_element('td', ['class' => "{$method}togglefield"], $toggle_link);
 			$i++;
 		}
+		
+		$group_row = elgg_format_element('td', ['class' => 'namefield'], elgg_format_element('div', [], $group->name));
```

Comment:
In this particular case, I don't have any preference, as it is messy no matter what you do with it. In general, I prefer to create a separate view for module body, as it ends up cleaner and allows me to use inline tags.